tests/tensor/test_slinalg.py

```diff
@@ -170,122 +174,110 @@ def test_eigvalsh_grad():
     )
 
 
-class TestSolve(utt.InferShapeTester):
-    def setup_method(self):
-        self.op_class = Solve
-        self.op = Solve()
-        super().setup_method()
-
-    def test_infer_shape(self):
-        rng = np.random.default_rng(utt.fetch_seed())
+class TestSolveBase(utt.InferShapeTester):
+    @pytest.mark.parametrize(
+        "A_func, b_func, error_message",
+        [
+            (vector, matrix, "`A` must be a matrix"),
+            (
+                functools.partial(tensor, dtype="floatX", broadcastable=(False,) * 3),
+                matrix,
+                "`A` must be a matrix",
+            ),
+            (
+                matrix,
+                functools.partial(tensor, dtype="floatX", broadcastable=(False,) * 3),
+                "`b` must be a matrix or a vector",
+            ),
+        ],
+    )
+    def test_make_node(self, A_func, b_func, error_message):
+        np.random.default_rng(utt.fetch_seed())
+        with pytest.raises(ValueError) as excinfo:
+            A = A_func()
+            b = b_func()
+            SolveBase()(A, b)
+        assert error_message in str(excinfo.value)
```

Comment:
FYI: the `match` keyword to `pytest.raises` can be used to verify the exception message more succinctly.

Reply:
The idea was incorporated to shorten the test